sources/api/schnauzer/src/helpers.rs

```diff
@@ -737,6 +749,98 @@ pub fn pause_prefix(
     Ok(())
 }
 
+/// The `tuf-prefix` helper is used to map an AWS region to the correct TUF
+/// repository.
+///
+/// This helper takes the AWS region as its only parameter, and returns the
+/// fully qualified domain name to the correct TUF repository.
+///
+/// # Fallback
+///
+/// A map of region to TUF repository bucket is maintained herein. But if we do
+/// not have the region in our map, a fallback repository is returned. This
+/// would allow a version of Bottlerocket to run in a new region before this map
+/// has been updated.
+///
+/// # Example
+///
+/// In this example the repository bucket for the region will be returned.
+/// `{{ tuf-prefix settings.aws.region }}`
+///
+/// This would result in something like:
+/// `https://bottlerocket-updates-us-west-2.s3.us-west-2.amazonaws.com/latest`
+pub fn tuf_prefix(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    trace!("Starting tuf helper");
+    let template_name = template_name(renderctx);
+    check_param_count(helper, template_name, 1)?;
+
+    // get the region parameter, which is probably given by the template value
+    // settings.aws.region. regardless, we expect it to be a string.
+    let aws_region = get_param(helper, 0)?;
+    let aws_region = aws_region.as_str().with_context(|| error::AwsRegionSnafu {
+        value: aws_region.to_owned(),
+        template: template_name,
+    })?;
+
+    // construct the registry fqdn
+    let tuf_repository = tuf_repository(aws_region);
+
+    // write it to the template
+    out.write(&tuf_repository)
+        .with_context(|_| error::TemplateWriteSnafu {
+            template: template_name.to_owned(),
+        })?;
+
+    Ok(())
+}
+
+/// The `metadata-prefix` helper is used to map an AWS region to the correct
+/// metadata location inside of the TUF repository.
+///
+/// This helper takes the AWS region as its only parameter, and returns the
+/// prefix of the metadata.
+///
+pub fn metadata_prefix(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    trace!("Starting tuf helper");
+    let template_name = template_name(renderctx);
+    check_param_count(helper, template_name, 1)?;
+
+    // get the region parameter, which is probably given by the template value
+    // settings.aws.region. regardless, we expect it to be a string.
+    let aws_region = get_param(helper, 0)?;
+    let aws_region = aws_region.as_str().with_context(|| error::AwsRegionSnafu {
+        value: aws_region.to_owned(),
+        template: template_name,
+    })?;
+
+    // construct the prefix
+    let metadata_location = if TUF_BUCKET_MAP.contains_key(aws_region) {
+        "/metadata"
+    } else {
+        ""
+    };
```

Comment:
nit: IMO the template would be slightly more clear if we omit the leading `/` here:
```suggestion
    // construct the prefix
    let metadata_location = if TUF_BUCKET_MAP.contains_key(aws_region) {
        "metadata"
    } else {
        ""
    };
```

So then the template would be:
```
{{ tuf-prefix settings.aws.region }}/{{ metadata-prefix settings.aws.region }}/2020-07-07/{{ os.variant_id }}/{{ os.arch }}/
```

Reply:
The problem there would be regions pointing to the current public repository would render as: `https://updates.bottlerocket.aws//2020-07-07/aws-k8s-1.23/x86_64/`.